improver/wxcode/modal_code.py

```diff
@@ -244,28 +315,432 @@ def process(self, cubes: CubeList) -> Cube:
+        """
+        # Find indices corresponding to dry weather codes inclusive of a wet bias.
+        dry_counts = np.sum(
+            np.isin(cube.data, self.broad_categories["dry"]), axis=time_axis
+        )
+        wet_counts = np.sum(
+            np.isin(cube.data, self.broad_categories["wet"]), axis=time_axis
+        )
+        return dry_counts > self.wet_bias * wet_counts
+
+    def _find_most_significant_dry_code(
+        self, cube: Cube, result: Cube, dry_indices: np.ndarray, time_axis: int
+    ) -> Cube:
+        """Find the most significant dry weather code at each point.
+
+        Args:
+            cube: Weather code cube.
+            result: Cube into which to put the result.
+            dry_indices: Boolean, which is true if the weather codes at that point,
+                are dry.
+            time_axis: The time coordinate dimension.
+
+        Returns:
+            Cube where points that are dry are filled with the most common dry
+            code present at that point. If there is a tie, the most significant dry
+            weather code is used, assuming higher values for the weather code indicate
+            more significant weather.
+        """
+        # Clip the weather codes to be within the range given by the dry weather codes.
+        cube_min = np.min(cube.data, axis=time_axis)
+        cube_max = np.max(cube.data, axis=time_axis)
+        min_clip_value = np.max(
+            [
+                np.broadcast_to(np.min(self.broad_categories["dry"]), cube_min.shape),
+                cube_min,
+            ],
+            axis=time_axis,
+        )
+        max_clip_value = np.min(
+            [
+                np.broadcast_to(np.max(self.broad_categories["dry"]), cube_max.shape),
+                cube_max,
+            ],
+            axis=time_axis,
+        )
+        uniques, counts = np.unique(
+            np.clip(cube.data, min_clip_value, max_clip_value),
+            return_counts=True,
+            axis=time_axis,
+        )
+        # Flip the unique values and the counts to be in descending order, so that
+        # the argmax will use the weather code with the lowest index in the event of
+        # a tie.
+        uniques = np.flip(uniques, axis=time_axis)
+        counts = np.flip(counts, axis=time_axis)
+        result.data[dry_indices] = uniques[np.argmax(counts)][dry_indices]
+        return result
+
+    def _find_non_intensity_indices(self, cube: Cube, time_axis: int) -> np.ndarray:
+        """Find which points have predictions for weather codes from any of the
+        intensity categories.
+
+        Args:
+            cube: Weather code cube.
+            time_axis: The time coordinate dimension.
+
+        Returns:
+            Boolean that is True is any weather code from the intensity categories
+            are found at a given point, otherwise False.
+        """
+        return ~np.sum(
+            np.isin(cube.data, self.intensity_categories.values()), axis=time_axis
```

Comment:
I don't believe this is working at present.
`self.intensity_categories.values()` looks like this:
`dict_values([[10, 14], [12, 15], [23, 26], [24, 27], [29, 30]])`
for which an `isin` test does not work, this needs to be flattened in someway (e.g. `np.array(list(intensity_categories.values())).flatten()`)  or iterated over the sub-groups.

Currently it will always return False from the `np.isin` statement.
Unless I've made a mistake in understanding this we should be concerned that the unit tests have not highlighted an issue.

Reply:
Amended to work as intended.